### PR TITLE
Update to meet CommandValue and CommandRequest refactoring

### LINKED ISSUE #12: Update to meet CommandValue and CommandRequest refactoring
Modify the related code to meet this refactoring: https://github.com/edgexfoundry/device-sdk-go/pull/262

### PR DESCRIPTION
**Pull request by FelixTing**
fix #12 #13 

Also update:
1. Recover and update the mkdir codes, PR#5 is my mistaken
2. Move mutex to handleReadCommands and handleWriteCommands to prevent
irregular runtime error when open db
3. Add float encoding Base64 and eNotation to device profile

Signed-off-by: Felix Ting <felix@iotechsys.com>

### CODE REVIEW

**Comment by FelixTing** [comment]
recheck

File: cmd/res/device.virtual.float.yaml [hunk 1]
```diff
@@ -7,7 +7,15 @@ description: "Example of Device-Virtual"
 
 deviceResources:
 -
-  name: "Enable_Randomization"
+  name: "EnableRandomization_Float32"
+  description: "used to decide whether to re-generate a random value"
+  properties:
+    value:
+      { type: "Bool", readWrite: "W", defaultValue: "true" }
```
Comment (cloudxxx8) [inline cmd/res/device.virtual.float.yaml]: why the type is Bool for Float32?

Comment (FelixTing) [reply cmd/res/device.virtual.float.yaml]: EnableRandomization_Float32 is used to decide whether to re-generate a random float32 value, so it's type is Boolean

**Comment by cloudxxx8** [comment]
LGTM

**Pull request merged**